### ISSUE TITLE
Restore __version__ to twisted/__init__.py on Python 3, once twisted.python.versions is enabled

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5886]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5886|
|Type|enhancement|
|Created|2012-08-19 19:45:24Z|

`twisted.__version__` should be enabled on Python 3 once `twisted.python.versions` has been ported.

<details><summary>Searchable metadata</summary>

```
trac-id__5886 5886
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1345405524000000 1345405524000000
changetime__1346241741000000 1346241741000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [35426]) re-enable twisted.__version__

refs [#5917](https://github.com/twisted/twisted/issues/5917)
refs [#5886](https://github.com/twisted/twisted/issues/5886)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `closed`|
|-|-|

(In [35431]) Merge versions-python3-5917-3

Author: exarkun
Reviewer: itamar
Fixes: [#5917](https://github.com/twisted/twisted/issues/5917)
Fixes: [#5785](https://github.com/twisted/twisted/issues/5785)
Fixes: [#5886](https://github.com/twisted/twisted/issues/5886)

Port `twisted.python.versions` to Python 3.  Also add a bit of missing test coverage
and restore the definition of `twisted.__version__` on Python 3.
